### DISCUSSION TITLE
Gatsby Site Requiring Hard Refresh to See New Posts

### DISCUSSION THREAD

**Discussion by kim-lindgren** [gatsbyjs/gatsby#35262] [Help] [answered]
This issue has already been brought up and in most cases seems to involve gatsby-plugin-offline.

I am NOT using gatsby-plugin-offline and I still have this problem. Whenever I push to the remote, my live site will update all changes I make except new blog posts, supposedly due to graphql not querying another time. But during production on my localhost it updates instantly.

I find it confusing to read all the different cases of this error, for instance in this thread it says that you SHOULD use gatsby-plugin-offline to solve the issue.
https://stackoverflow.com/questions/67725130/gatsby-site-requires-refresh-to-view-new-data

It shows how to create a popup whenever the site needs to refresh. Is there really no other way? Is it not possible to simply have a Gatsby site work like a normal website, automatically showing the latest blog posts?

**Comment by LekoArts** [comment, marked as answer]
You'll need to make sure that your caching is set up correctly: https://www.gatsbyjs.com/docs/caching/
Gatsby sites don't require a hard refresh when updates are pushed.

**Reply by kim-lindgren** [reply]
Thank you.

I have never worked with caching before, so the docs you sent don't tell me very much. Like, suppose I want to disable caching for the markdown blog posts specifically (src/blog), what do I do? I use Firebase for hosting.

**Reply by LekoArts** [reply]
You wouldn't want to disable caching for specific pages/posts but set up caching rules for the files that are created. https://www.gatsbyjs.com/docs/how-to/previews-deploys-hosting/deploying-to-firebase/ has explanations on what to put into your `firebase.json` for correct caching headers

**Reply by kim-lindgren** [reply]
Thanks, the default settings work fine (my backend developer had created some other settings in that file). Problem solved!

**Comment by hesta-inrsaurabh** [comment]
I deployed on build on  digital ocean, facing this issue, on HARD REFRESH new data shows, but for some users, it keeps showing older data.
